### DISCUSSION TITLE
how to build "API" with "ID" parameters using app.Dao()

### DISCUSSION THREAD

**Discussion by ejrua** [pocketbase/pocketbase#613] [Q&A]
how to build "API" with "ID" parameters using app.Dao()

**Comment by ganigeorgiev** [comment]
Sorry, I'm not sure that I understand the question.

Could you provide a little more info on what are you trying to do?

**Comment by ejrua** [comment]
thank,  with "ID" parameters, code example complet, please 

I don't know how to make the "API" with parameters for the "query".

e.Router.AddRoute(echo.Route{
			Method: http.MethodGet,
			Path:   "/api/mio",
			Handler: func(c echo.Context) error {

........


queryErr := app.Dao().DB().	
					NewQuery(`select ie.nombre,pp.descripcion,count(1)
                                                            from docentes_movimientos dm,
                                                                    docentes d,
                                                                    partido_politico pp,
                                                                    institucion_educativa ie
                                                            where dm.docente_id = d.id 
                                                                and dm.partido_politico_id = pp.id
                                                                and ie.id = d.institucion_educativa_id
                                                                and  ie.id = ? 
                                                                and asociado = 1
                                                            group by d.institucion_educativa_id,pp.id`).
All(&result)
				if queryErr != nil {
					return rest.NewBadRequestError("Failed to fetch comments count.", queryErr)
				}

**Reply by ganigeorgiev** [reply]
You want to pass an id from a route path to a custom query, right?

If that's the case:

1. Change your path to `/api/mio/:id` (path parameters are defined with `:`)

2. In the raw sql I recommend using a named parameter - replace `?` with `{:id}`

3. Bind the path parameter to the query: `q.Bind(dbx.Params{"id": c.PathParam("id")})`

Or in other words, your query should look something like this:
```go
result := []struct{
	Nombre string
	Descripcion string
	Total int
}{}

query := app.Dao().DB().NewQuery(`
	SELECT ie.nombre, pp.descripcion, count(*) as total
	FROM docentes_movimientos dm, docentes d, partido_politico pp, institucion_educativa ie
	WHERE dm.docente_id = d.id and
		dm.partido_politico_id = pp.id and
		ie.id = d.institucion_educativa_id and
		ie.id = {:id} and
		asociado = 1
	group by d.institucion_educativa_id,pp.id
`).Bind(dbx.Params{"id": c.PathParam("id")})


if err := query.All(&result); err != nil {
	return rest.NewBadRequestError("Failed to fetch comments count.", err)
}
```

**Reply by ganigeorgiev** [reply]
_footnote: I don't fully understand what the above query is supposed to do, but it may be easier instead of executing a custom query to do several small requests (you can get the count from the paginated result for a collection)._

**Reply by ejrua** [reply]
thanks, excellent example, with data to graph "chart.js" with client-side SvelteKit
